cogs/poll.py

```diff
@@ -1,131 +1,199 @@
 import asyncio
+
 import discord
+from discord import Message, Embed
 from discord.ext import commands
+from discord.ext.commands import Context
 from cogs.base import BaseCog
 
+MAX = 10
+MIN = 1
+
 
 class Poll(BaseCog):
 
     @commands.command(pass_context=True)
-    async def poll(self, ctx, question, minutes, *options: str):
-        """ Generates a poll. Consists of a question, the options to choose from and the number of minutes to keep
+    async def poll(self, ctx: Context, question: str, minutes: str, *options: str) -> None:
+        """
+        generates a poll. Consists of a question, the options to choose from and the number of minutes to keep
         the poll going
 
-        :param ctx: Context of the command message
-        :param question: Question for the poll
-        :param options: List of options to choose from
-        :param minutes: Number of minutes to run the poll for
-        :return: None
+        Parameters
+        ----------
+        ctx : Context
+        question : Str
+        minutes : Str
+        options : Str
+
+        Returns
+        -------
+        None
         """
-        if number_of_options_does_not_fall_in_range(options):
-            await ctx.send('Number of options needs to be more than 1 and less than 10. Number of options given: {}.'
-                           .format(len(options)))
+        if MIN >= len(options) or len(options) > MAX:
+            await ctx.message.delete()
```

Comment:
This should probably be the first thing we do before a check. This line is also duplicated on 37.

Reply:
Why are we deleting the message before or after we check if it's valid?

Reply:
The message is the command such as "!poll What_game? 1 Divinity Minecraft DND" just so the ugly command message goes away.

Reply:
So, do whichever you prefer, but my personal preference is to leave the original command a user posted. It helps to show others how to use the same command and can help a user understand what they did wrong if the result was an error or not what they expected.

Analogy: I wouldn't want my bash commands getting deleted when they execute so I couldn't see how I invoked it if something messes up or if I want to do it again.